omnigibson/transition_rules.py

```diff
@@ -1266,10 +1274,13 @@ def _execute_recipe(cls, container, recipe, in_volume):
                 system.remove_all_group_particles(group=group_name)
 
         # Remove either all objects or only the recipe-relevant objects inside the container
-        objs_to_remove.extend(np.concatenate([
-            cls._OBJECTS[np.where(in_volume[cls._CATEGORY_IDXS[obj_category]])[0]]
-            for obj_category in recipe["input_objects"].keys()
-        ]) if cls.ignore_nonrecipe_objects else cls._OBJECTS[np.where(in_volume)[0]])
+        object_mask = in_volume.copy()
```

Comment:
i think this was the logic fix from awhile ago, right @cgokmen ? what is this doing exactly?

Reply:
Something was wrong with the indexing here (in fact dimensions were not matching etc) this fixes that.